### PR TITLE
feat: Add LookUpTable helper

### PR DESCRIPTION
**Pull request by afonsofonbraga**
on top of https://github.com/tidewise/drivers-radar_base/pull/1

These files were imported from the radar temporary repo.  Added

### CODE REVIEW

File: src/EchoToImageLUT.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    : m_num_angles(num_angles)
+    , m_sweep_size(sweep_size)
+{
+    cout << "Setting Lookup table up\n";
+    setup(sweep_size, window_size, beam_width);
+}
+
+void EchoToImageLUT::setup(int sweep_size, int window_size, float beam_width)
+{
+    auto raw_data = computeRawLUTTable(sweep_size, window_size, m_num_angles, beam_width);
```
Comment (jhonasiv) [inline src/EchoToImageLUT.cpp]: `m_num_angles` is only used here, I think you could make it an argument of the `setup` method instead.

File: src/EchoToImageLUT.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            int sweep_size,
+            int num_angles,
+            cv::Point2i const& p);
+
+        void setup(int sweep_size, int window_size, float beam_width);
+
+    public:
+        EchoToImageLUT() = delete;
+
+        EchoToImageLUT(int num_angles, int sweep_size, float beam_width, int window_size);
```
Comment (jhonasiv) [inline src/EchoToImageLUT.hpp:44]: Add some documentation explaining the parameters. `num_angles` and `beam_width` might seem weird if some radar doesnt return data like Furuno, so make sure it's clear.

File: src/EchoToImageLUT.cpp (hunk 1 above)
Comment (afonsofonbraga) [reply src/EchoToImageLUT.cpp]: I didn't touch this file, just imported from the temporary repo. I'll take a look into it next week.

File: src/EchoToImageLUT.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            m_data.push_back(table[global_echo_id][point_id]);
+        }
+    }
+    m_data_index[table.size()] = m_data.size();
+}
+
+int EchoToImageLUT::normalizeAngle(int angle, int num_angles)
+{
+    if (angle < 0) {
+        return angle + num_angles;
```
Comment (kapeps) [inline src/EchoToImageLUT.cpp:84]: shouldn't it be angle%num_angles to normalize it?

**Review by kapeps — APPROVED** [review]

File: src/EchoToImageLUT.hpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+         * @param window_size The param will create a squared frame.
+         */
+        EchoToImageLUT(int num_angles, int sweep_size, float beam_width, int window_size);
+        ~EchoToImageLUT() = default;
+
+        void updateImage(cv::Mat& image, int angle, int echo_index, int echo) const;
+
+        void updateAngle(cv::Mat& image, int angle, int echo_index, int echo) const;
+
+        void clearBeamEndAtImage(cv::Mat& image, int angle, int echo_index) const;
```
Comment (jhonasiv) [inline src/EchoToImageLUT.hpp]: This isnt implemented

File: src/EchoToImageLUT.hpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+         * @param sweep_size Number of measurements inside a single sweep.
+         * @param beam_width Angle between the 3dB half power points on the main lobe.
+         * @param window_size The param will create a squared frame.
+         */
+        EchoToImageLUT(int num_angles, int sweep_size, float beam_width, int window_size);
+        ~EchoToImageLUT() = default;
+
+        void updateImage(cv::Mat& image, int angle, int echo_index, int echo) const;
+
+        void updateAngle(cv::Mat& image, int angle, int echo_index, int echo) const;
```
Comment (jhonasiv) [inline src/EchoToImageLUT.hpp]: Not implemented

**Review by jhonasiv — APPROVED** [review]

**Pull request merged**